kong/reports.lua

```diff
@@ -303,6 +303,12 @@ function get_current_suffix(ctx)
     return nil
   end
 
+  -- 400 case is for invalid requests, eg: if a client sends a HTTP
+  -- request to a HTTPS port, it does not initialized any Nginx variables
+  if kong.response.get_status() == 400 and proxy_mode == "" then
```

Comment:
I  would first check the proxy mode:
```suggestion
  if proxy_mode == "" and kong.response.get_status() == 400 then
```

Reply:
done